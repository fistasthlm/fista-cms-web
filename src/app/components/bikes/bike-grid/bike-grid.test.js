import React from 'react';
import { shallow } from 'enzyme';
import { Map, List } from 'immutable';
import BikeGrid from './bike-grid';

const setup = propOverrides => {
    const props = Object.assign({
        bikes: List(),
    }, propOverrides);

    return shallow(<BikeGrid {...props} />);
};

describe('Render', () => {
    it('should render component', () => {
        const wrapper = setup();

        expect(wrapper).toMatchSnapshot();
    });

    it('should render grid', () => {
        const wrapper = setup({
            bikes: List([
                Map({
                    _id: 'yolo',
                    title: 'dolan',
                })
            ])
        });

        expect(wrapper).toMatchSnapshot();
    });
});
