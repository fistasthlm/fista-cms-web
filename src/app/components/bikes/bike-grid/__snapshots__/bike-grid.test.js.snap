// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="bike-grid"
/>
`;

exports[`Render should render grid 1`] = `
<div
  className="bike-grid"
>
  <BikeTile
    bike={
      Immutable.Map {
        "_id": "yolo",
        "title": "dolan",
      }
    }
    key="yolo"
  />
</div>
`;
